com.unity.netcode.adapter.utp/Runtime/UnityTransport.cs

```diff
@@ -517,13 +517,67 @@ private void Update()
                 {
                     ;
                 }
+                
+#if MULTIPLAYER_TOOLS
+                if (NetworkManager.Singleton.IsServer)
+                {
+                    var ngoConnectionIds = NetworkManager.Singleton.ConnectedClients.Keys;
+                    foreach (var ngoConnectionId in ngoConnectionIds)
+                    {
+                        ExtractNetworkMetrics(NetworkManager.Singleton.ClientIdToTransportId(ngoConnectionId));
+                    }
+                }
+                else
+                {
+                    ExtractNetworkMetrics(NetworkManager.Singleton.ClientIdToTransportId(NetworkManager.Singleton.ServerClientId));
+                }
+#endif
             }
         }
 
         private void OnDestroy()
         {
             DisposeDriver();
         }
+        
+#if MULTIPLAYER_TOOLS
+        private void ExtractNetworkMetrics(ulong transportClientId)
+        {
+            var networkConnection =  ParseClientId(transportClientId);
+            
+            //Two pipelines, unreliable sequence and reliable fragmented
+            for (byte i = 0; i < 2; ++i)
+            {
+                var pipeline = i == 0 ? m_UnreliableSequencedPipeline : m_ReliableSequencedFragmentedPipeline;
+
+#if ENABLE_UNITY_COLLECTIONS_CHECKS
```

Comment:
Is this an existing pattern? Definitely gets the job done, a little awkward though